module/os/freebsd/zfs/vdev_geom.c

```diff
@@ -1053,7 +1053,7 @@ vdev_geom_io_intr(struct bio *bp)
 	/*
 	 * We have to split bio freeing into two parts, because the ABD code
 	 * cannot be called in this context and vdev_op_io_done is not called
```

Comment:
With this PR `vdev_op_io_done` _is_ now called for `ZIO_TYPE_FLUSH`, so this comment is wrong. I didn't want to just remove it, because maybe its pointing to a cleanup opportunity if this is no longer necessary. But I also don't know enough right now to just make the change. I'll come back to it soon, but if a FreeBSDer wants to step in and propose a change, that'd be even better!

Reply:
This context is indeed not sleepable.  Primarily it means no waiting memory allocations, etc, which would not be a problem here.  But it also means no sleepable locks, and on FreeBSD ZFS' mutex_enter() called, for example, by zfs_refcount_remove_many(), called by abd_return_buf() is mapped into sleepable sx_xlock().  Since vdev_op_io_done() is now called for ZIO_TYPE_FLUSH, I guess it should be easier to remove this special case here and make vdev_geom_io_done() do everything.